ipa-core/src/report/hybrid.rs

```diff
@@ -162,6 +409,21 @@ impl UniqueBytes for EncryptedHybridReport {
     }
 }
 
+impl<BK, B> UniqueBytes for EncryptedHybridImpressionReport<BK, B>
+where
+    B: Deref<Target = [u8]>,
+    BK: SharedValue,
+    Replicated<BK>: Serializable,
+    <Replicated<BK> as Serializable>::Size: Add<U16>,
+    <<Replicated<BK> as Serializable>::Size as Add<U16>>::Output: ArrayLength,
+{
+    /// We use the `TagSize` (the first 16 bytes of the ciphertext) for collision-detection
+    /// See [analysis here for uniqueness](https://eprint.iacr.org/2019/624)
+    fn unique_bytes(&self) -> Vec<u8> {
```

Comment:
I think it would be more flexible to return `&[u8]` as it would work for both cases: when you need to peek into those bytes and when you need to clone them. This API always clones which may not be desirable

Reply:
Makes sense. I'll change it